modules/oke/nodepools.tf

```diff
@@ -21,11 +21,15 @@ resource "oci_containerengine_node_pool" "nodepools" {
       }
     }
     # set quantity to a minimum of 1 to allow small clusters. 
-    size = max(1, element(each.value,1))
+    size = max(1, element(each.value,2))
   }
+  node_shape_config {
```

Comment:
see if you can use a dynamic here like thus: https://github.com/oracle-terraform-modules/terraform-oci-vcn/blob/master/nat.tf#L27

so the node_shape_config will only be evaluated iff the shape is flex.